mirdata/datasets/idmt_smt_audio_effects.py

```diff
@@ -0,0 +1,228 @@
+        fx_setting (int): effect setting number
+
+
+    Cached Properties:
+        annotation (EventData): a description of this annotation
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+
+    @property
+    def instrument(self):
+        return self._track_metadata.get("instrument")
+
+    @property
+    def midi_nr(self):
+        return self._track_metadata.get("midi_nr")
+
+    @property
+    def fx_group(self):
+        return self._track_metadata.get("fx_group")
+
+    @property
+    def fx_type(self):
+        return self._track_metadata.get("fx_type")
+
+    @property
+    def fx_setting(self):
+        return self._track_metadata.get("fx_setting")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The track's audio
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+        Returns:
+            jams.JAMS: the track's data in jams format
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata=self._track_metadata,
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load an The IDMT-SMT-Audio Effect track
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+    """
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The IDMT-SMT-Audio Effect dataset
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="idmt_smt_audio_effects",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            download_info=DOWNLOAD_INFO,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_index = {
+            "fileID": {
+                "list_id": str,
+                "instrument": str,
+                "midi_nr": str,
+                "fx_group": int,
+                "fx_type": int,
+                "fx_setting": int,
+            }
+        }
+
+        for root, dirs, files in os.walk(self.data_home):
+            for file in files:
+                if file.endswith(".xml"):
+                    xml_path = os.path.join(root, file)
+                    tree = ET.parse(xml_path)
```

Comment:
You would need to add something like that here:
```
try:
    tree = ET.parse(xml_path)
except:
   except FileNotFoundError:
   raise FileNotFoundError("Metadata file for not found. Did you run .download()?")
```
Maybe in your case it would be nice if you add a little stentence in the error message that you get the metadata from multiple files and that if there is one missing the error is gonna show as well.